release/storage/aws.py

```diff
@@ -6,32 +6,38 @@
 from release.storage import AbstractStorageProvider
 
 
-def get_session(boto3_profile=None, region_name=None, access_key_id=None, secret_access_key=None):
-        if boto3_profile:
-            if access_key_id or secret_access_key or region_name:
-                raise ValueError("access_key_id, secret_access_key, and region_name cannot be used with boto3_profile")
-            return boto3.session.Session(profile_name=boto3_profile)
-        elif access_key_id or secret_access_key or region_name:
-            if not access_key_id or not secret_access_key or not region_name:
-                raise ValueError("access_key_id, secret_access_key, and region_name must all be set")
-            return boto3.session.Session(
-                aws_access_key_id=access_key_id,
-                aws_secret_access_key=secret_access_key,
-                region_name=region_name)
-        else:
-            raise ValueError("boto_profile or explicit AWS credentials (access_key_id, "
-                             "secret_access_key, region_name) must be provided")
+def get_aws_session(access_key_id, secret_access_key, region_name=None):
+    """ This method will replace access_key_id and secret_access_key
+    with None if one is set to '' This allows falling back to the AWS internal
+    logic so that AWS_SESSION_TOKEN or something else can be used
```

Comment:
What does `something else` refer to here?
`some else like IAM role` ?

Reply:
or link the `http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials` again.